workshops/views.py

```diff
@@ -1628,41 +1628,52 @@ def all_trainings(request):
 
 
 @login_required
-def instructors(request):
-    '''Search for instructors.'''
+def workshop_staff(request):
+    '''Search for workshop staff.'''
     instructor_badges = Badge.objects.instructor_badges()
-    instructors = Person.objects\
-                        .filter(badges__in=instructor_badges) \
-                        .filter(airport__isnull=False) \
-                        .annotate(is_swc_instructor=Sum(
-                                      Case(When(badges__name='swc-instructor',
-                                                then=1),
-                                           default=0,
-                                           output_field=IntegerField())),
-                                  is_dc_instructor=Sum(
-                                      Case(When(badges__name='dc-instructor',
-                                                then=1),
-                                           default=0,
-                                           output_field=IntegerField()))) \
-                        .select_related('airport') \
-                        .prefetch_related('lessons')
-    instructors = instructors.annotate(
+    TTT = Tag.objects.get(name='TTT')
+    stalled = Tag.objects.get(name='stalled')
+
+    people = Person.objects.filter(airport__isnull=False) \
+                           .select_related('airport') \
+                           .prefetch_related('badges', 'lessons')
+
+    trainees = Task.objects.filter(event__tags=TTT) \
+                           .filter(role__name='learner') \
+                           .filter(person__airport__isnull=False) \
+                           .exclude(event__tags=stalled) \
+                           .exclude(person__badges__in=instructor_badges) \
+                           .values_list('person__pk', flat=True)
+
+    # we need to count number of specific roles users had
+    # and if they are SWC/DC instructors
+    people = people.annotate(
         num_taught=Count(
             Case(
-                When(
-                    task__role__name='instructor',
-                    then=Value(1)
-                ),
+                When(task__role__name='instructor', then=Value(1)),
+                output_field=IntegerField()
+            )
+        ),
```

Comment:
@gvwilson, what about an instructor who taught at one SWC workshop and two TTT? Do we want to display `1`, `2` or `3`?